### PR TITLE
[IOTDB-3416] Abstract interface of ConfigTask Execution for standalone IoTDB

### PR DESCRIPTION
**Pull request by RYH61**
图片

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/IConfigTaskFetcher.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import org.apache.iotdb.db.mpp.plan.statement.metadata.SetStorageGroupStatement;
+import org.apache.iotdb.db.mpp.plan.statement.metadata.SetTTLStatement;
+import org.apache.iotdb.db.mpp.plan.statement.metadata.ShowStorageGroupStatement;
+import org.apache.iotdb.db.mpp.plan.statement.metadata.ShowTTLStatement;
+
+import com.google.common.util.concurrent.SettableFuture;
+
+import java.util.List;
+
+public interface IConfigTaskFetcher {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/IConfigTaskFetcher.java:35]:
```suggestion
public interface IConfigTaskExecutor {
```

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/ClusterConfigTaskFetcher.java [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ClusterConfigTaskFetcher implements IConfigTaskFetcher {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/ClusterConfigTaskFetcher.java:65]:
```suggestion
public class ClusterConfigTaskExecutor implements IConfigTaskFetcher {
```

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/StandsloneConfigTaskFetcher.java [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class StandsloneConfigTaskFetcher implements IConfigTaskFetcher {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/StandsloneConfigTaskFetcher.java:56]:
```suggestion
public class StandsloneConfigTaskExecutor implements IConfigTaskFetcher {
```

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/IConfigTaskFetcher.java (hunk 1 above)
Comment (RYH61) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/IConfigTaskFetcher.java]: fixed~

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/ClusterConfigTaskFetcher.java (hunk 2 above)
Comment (RYH61) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/ClusterConfigTaskFetcher.java]: fixed~

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/StandsloneConfigTaskFetcher.java (hunk 3 above)
Comment (RYH61) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/StandsloneConfigTaskFetcher.java]: fixed~

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/ShowTTLTask.java [hunk 4]
```diff
@@ -61,63 +41,13 @@ public ShowTTLTask(ShowTTLStatement showTTLStatement) {
...
-                  allStorageGroupToTTL.get(matchedStorageGroupPath));
-            }
-          }
-        }
-      } catch (MetadataException e) {
-        future.setException(e);
-      }
-    }
-    // build TSBlock
+    return configTaskExecutor.showTTL(showTTLStatement);
```
Comment (mychaow) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/ShowTTLTask.java:46]: show TTL?

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/ClusterConfigTaskExecutor.java [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ClusterConfigTaskExecutor implements IConfigTaskExecutor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(StandsloneConfigTaskExecutor.class);
```
Comment (mychaow) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/ClusterConfigTaskExecutor.java:67]: `ClusterConfigTaskExecutor`

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/ClusterConfigTaskExecutor.java [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+
+  private static final IClientManager<PartitionRegionId, ConfigNodeClient>
+      CONFIG_NODE_CLIENT_MANAGER =
+          new IClientManager.Factory<PartitionRegionId, ConfigNodeClient>()
+              .createClientManager(new DataNodeClientPoolFactory.ConfigNodeClientPoolFactory());
+
+  private static final class ClusterConfigTaskFetcherHolder {
```
Comment (mychaow) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/ClusterConfigTaskExecutor.java:74]: `ClusterConfigTaskExecutorHolder`

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/IConfigTaskExecutor.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.mpp.plan.execution.config.fetcher;
```
Comment (mychaow) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/IConfigTaskExecutor.java:20]: fetcher -> executor?

**Review by mychaow — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/ClusterConfigTaskExecutor.java (hunk 5 above)
Comment (RYH61) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/ClusterConfigTaskExecutor.java]: fixed~

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/ClusterConfigTaskExecutor.java (hunk 6 above)
Comment (RYH61) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/ClusterConfigTaskExecutor.java]: fixed~

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/IConfigTaskExecutor.java (hunk 7 above)
Comment (RYH61) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/execution/config/fetcher/IConfigTaskExecutor.java]: fixed~

**Review by qiaojialin — APPROVED** [review]

**Review by mychaow — APPROVED** [review]
LGTM

**Pull request merged**